rust_src/src/windows.rs

```diff
@@ -95,7 +95,7 @@ impl LispWindowRef {
 
         if mode_line_height >= 0 {
             mode_line_height
```

Comment:
Why did you change from `!=` to `< 0`? The C code used:
```
#define CURRENT_MODE_LINE_HEIGHT(W)					\
  (W->mode_line_height >= 0						\
   ? W->mode_line_height						\
   : (W->mode_line_height						\
      = (MATRIX_MODE_LINE_HEIGHT (W->current_matrix)			\
	 ? MATRIX_MODE_LINE_HEIGHT (W->current_matrix)			\
	 : estimate_mode_line_height					\
	     (XFRAME (W->frame), CURRENT_MODE_LINE_FACE_ID (W)))))
```

Which means `MATRIX_MODE_LINE_HEIGHT` is only checked for being not 0. -1 and 1 are both true values as far as `if` is concerned. Only `0` is false.

Reply:
I don't know, I must have misread something.

Reply:
I read it twice myself. The variable checked flips from the if to the else if which adds to the confusion.

Reply:
This is fixed btw (that is, the change was removed entirely).